File: tests/fixtures/rust_project/src/services.rs
//! Service implementations.

use crate::models::User;
use crate::traits::{Validate, Summary};

impl User {
    /// Create a new user.
    pub fn new(name: &str, age: u32) -> Self {
        Self {
            name: name.to_string(),
            age,
        }
    }

    /// Check if the user is an adult.
    pub fn is_adult(&self) -> bool {
        self.age >= 18
    }

    fn internal_helper(&self) -> String {
        self.name.to_lowercase()
    }
}

impl Validate for User {
    fn is_valid(&self) -> bool {
        !self.name.is_empty() && self.age < 200
    }
}

impl Summary for User {
    fn summarize(&self) -> String {
        format!("{} (age {})", self.name, self.age)
    }
}

/// Process a batch of users.
pub fn process_users(users: &[User]) -> Vec<String> {
    users.iter().map(|u| u.summarize()).collect()
}

/// A duplicate symbol name (also exists in utils/helpers.rs).
pub fn helper() -> &'static str {
    "services helper"
}

File: tests/fixtures/python_project/src/services.py
"""Service module that imports from other modules."""

from models import User
from utils import format_name

def process_user(name: str, age: int) -> str:
    """Create and process a user."""
    user = User(name, age)
    if user.is_adult():
        return format_name(name, "verified")
    return format_name(name, "pending")

def list_users(users: list) -> list:
    """List all user names."""
    return [format_name(u.name, "") for u in users]
